File: 523-continuous-subarray-sum/continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        mapp = {0: -1}  # Dictionary to store remainder and its first occurrence index
        preSum = 0
        
        for index, num in enumerate(nums):
            preSum += num
            remainder = preSum % k
            
            if remainder in mapp:
                # Check if the subarray length is at least 2
                if index - mapp[remainder] > 1:
                    return True
            else:
                # Store the first occurrence of the remainder
                mapp[remainder] = index
        
        return False

File: 523-continuous-subarray-sum/continuous-subarray-sum.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        map.put(0, -1); // hashmap with remainder and index as keys, and values
        int prefixSum = 0;

        for (int i = 0; i < nums.length; i++){
            prefixSum += nums[i];

            int remainder = prefixSum % k;

            if (map.containsKey(remainder)) {
                if (i - map.get(remainder) > 1) {
                    return true;
                }
            } else {
                map.put(remainder, i);
                }
        }
        return false;   
    }
}